### ISSUE TITLE
Google signin on flowdash.co not working 

### ISSUE THREAD

**Issue by mammuth** [onejgordon/flow-dashboard#34]
When hitting sign in to initially create an account, the Google dialog opens and after sending it nothing happens. Same when repeating the sign in.

**Comment by onejgordon** [comment]
Thanks @mammuth -- what browser are you using? If you're able to bring up a JS console and paste any warnings or errors seen there, that may also be useful.

**Comment by mammuth** [comment]
Hm, tested it on desktop Chrome as well as Android chrome. On the desktop chrome there was an error with an extension which seems to cause the issue. 

However I don't know why it still isn't working on Android. But I guess this is no general issue, so maybe rather close the issue?

**Comment by PaulBGD** [comment]
Firefox, all console output:
```

          Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.  build.js:137334:9
Warning: RouterContext: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.  build.js:137334:9
"Download the React DevTools for a better development experience: https://fb.me/react-devtools"  build.js:111959:9
uncaught exception: [object Object]  (unknown)
Object { error: "popup_closed_by_user" }  build.js:145706:13
```

**Comment by onejgordon** [comment]
@PaulBGD any chance there's a popup blocker that's preventing signin? I just tested in Firefox 43.0.1 on a mac, and sign in completed.

**Comment by StephenAshmore** [comment]
Just to chime in here, this occurs to me only on my Android device running Chrome. I'll take a look at the console in the morning to see if it provides anything more helpful. When I click login, it opens a new tab which allows me to choose my email account. After that, it closes the tab and takes me back to flowdash.co ; however the page does not reload and you are not logged in.

It could be related to me having two Google Accounts on my phone, and having to choose which one. I'm choosing the one that I already have an account for(which works great on all of my non-mobile devices).

**Comment by Dinoshauer** [comment]
This happens to me as well on MacOS 10.12.4 with Chrome Version 58.0.3029.96 (64-bit) with uBlock origin off - Console output is:

    warning.js:36 Warning: RouterContext: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.
    printWarning @ warning.js:36
    warning @ warning.js:60
    createClass @ ReactClass.js:628
    (anonymous) @ RouterContext.js:43
    /Users/jeremygordon/pyprojects/flow/node_modules/react-router/lib/RouterContext.js../RouteUtils @ RouterContext.js:156
    s @ _prelude.js:1
    (anonymous) @ _prelude.js:1
    (anonymous) @ Router.js:29
    /Users/jeremygordon/pyprojects/flow/node_modules/react-router/lib/Router.js../InternalPropTypes @ Router.js:224
    s @ _prelude.js:1
    (anonymous) @ _prelude.js:1
    /Users/jeremygordon/pyprojects/flow/node_modules/react-router/lib/index.js../History @ index.js:39
    s @ _prelude.js:1
    (anonymous) @ _prelude.js:1
    /Users/jeremygordon/pyprojects/flow/src/js/main.js.babel/polyfill @ client_secrets.js:7
    s @ _prelude.js:1
    e @ _prelude.js:1
    (anonymous) @ _prelude.js:1
    cb=gapi.loaded_0:71 Uncaught Object {error: "idpiframe_initialization_failed", details: "Failed to read the 'localStorage' property from 'Window': Access is denied for this document."}
    (anonymous) @ cb=gapi.loaded_0:71
    Splash.js:20 Object {error: "popup_closed_by_user"}

**Comment by onejgordon** [comment]
Found this setting that can result in `Failed to read the 'localStorage' property from 'Window': Access is denied for this document.`

https://www.chromium.org/for-testers/bug-reporting-guidelines/uncaught-securityerror-failed-to-read-the-localstorage-property-from-window-access-is-denied-for-this-document

Could anyone experiencing this problem see if `Block third-party cookies and site data` is enabled?

**Comment by toddsiegel** [comment]
I was having this issue too. I have Privacy Badger installed and it was blocking cookies from apis.google.com and accounts.google.com. Unblocking them fixed it, but oddly this causes me issues nowhere else.

**Comment by onejgordon** [comment]
Looks to be related to Flow's use of the google-api-javascript-client. According to this issue, https://github.com/google/google-api-javascript-client/issues/260, best we can do for now is catch the error and show an informative toast to the user.

**Comment by xtensions** [comment]
@StephenAshmore 

Its happening in same way for my app. Working fine on chrome, safari on my mac, but not on android device, getting the same alert in error handling {"error","popup_closed_by_user"} right after selecting an account from list. 

Have you solved your issue.

**Comment by StephenAshmore** [comment]
@xtensions Yes, this began working for me. Now, the page reloads after selecting a google account.  The only thing I can think of for that would be perhaps a newer version of Chrome on Android or some software that could be blocking that API/popup. I'm running Chrome 62.0.3202.84.

**Comment by xtensions** [comment]
Thanks for your reply, it works for me after changing single_host_origin to none in api call.
